### PR TITLE
[SPARK-45257][CORE] Enable `spark.eventLog.compress` by default

### PR DESCRIPTION
**Pull request by dongjoon-hyun**
### What changes were proposed in this pull request?

This PR aims to enable `spark.eventLog.compress` by default for Apache Spark 4.0.0.

### Why are the changes needed?

- To save the event log storage cost by compressing the logs with ZStandard codec by default

### Does this PR introduce _any_ user-facing change?

Although we added a migration guide, the old Spark history servers are able to read the compressed logs.

### How was this patch tested?

 Pass the CIs.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Review by mridulm — APPROVED** [review]
We should have done this long ago !

**Comment by dongjoon-hyun** [comment]
Thank you for review and approval, @mridulm ! 😄

**Review by yaooqinn — APPROVED** [review]
LGTM, just one minor revison

**Review by ulysses-you — APPROVED** [review]

File: docs/core-migration-guide.md [hunk 1]
```diff
@@ -22,6 +22,10 @@ license: |
 * Table of contents
 {:toc}
 
+## Upgrading from Core 3.4 to 4.0
+
+- Since Spark 4.4, Spark will compress event logs. To restore the behavior before Spark 4.0, you can set `spark.eventLog.compress` to `false`.
```
[the remark this answers is not in the document]
Comment (dongjoon-hyun) [reply docs/core-migration-guide.md:27]: Oops! Thank you, @yaooqinn .

**Comment by dongjoon-hyun** [comment]
The last commit is a doc-only change and the previous commit passed `core` module unit tests.
Screenshot 2023-09-21 at 8 08 12 PM

Thank you all! Merged to master.

**Pull request closed without merging**